### PR TITLE
ENH: Add `subset` and `subsets` methods to `DisjointSet` class.

### PR DESCRIPTION
**Pull request by pmla**
#### What does this implement/fix?
This PR adds methods for getting subsets in the `DisjointSet` class.

@charris the logic is based on the code you kindly shared. Can I interest you in taking a look?

### CODE REVIEW

**Comment by sethtroisi** [comment]
I'll try to take a look at this tomorrow.

**Review by sethtroisi — APPROVED** [review]

**Comment by sethtroisi** [comment]
Build failure is unrelated (FAILED ../../sparse/linalg/eigen/arpack/tests/test_arpack.py::test_symmetric_modes).

I'd support this being merging in ~24 hour if there are no other comments.

File: scipy/_lib/_disjoint_set.py [hunk 1]
```diff
@@ -69,11 +71,22 @@ class DisjointSet:
...
+    Get all subsets in the disjoint set:
+
+    >>> disjoint_set.subsets()
+    [{1, 2}, {'a', 3, 'b'}]
     """
     def __init__(self, elements=None):
         self.n_subsets = 0
         self._sizes = {}
         self._parents = {}
+        self._nbrs = {}
```
[the remark this answers is not in the document]
Comment (pmla) [reply scipy/_lib/_disjoint_set.py:90]: Thanks Seth. I have added a comment explaining that it is a circular linked list.

**Comment by sethtroisi** [comment]
Thanks @pmla, this is a nice helper function for disjoint set

**Comment by pmla** [comment]
Thanks for the review and merge @sethtroisi

**Pull request merged**